src/lib.rs

```diff
@@ -98,6 +99,35 @@ impl<Data: AsRef<[u8]>> Reader<Data> {
             .unwrap()
             .map(move |level| &self.input.as_ref()[level.offset as usize..(level.offset + level.length_bytes) as usize])
     }
+
+    pub fn supercompression_global_data(&self) -> &[u8] {
+        let header = self.header();
+        let start = header.sgd_byte_offset as usize;
+        let end = (header.sgd_byte_offset + header.sgd_byte_length) as usize;
+        self.input.as_ref()[start..end].try_into().unwrap()
+    }
+
+    pub fn data_format_descriptors(&self) -> Vec<BasicDataFormatDescriptor> {
```

Comment:
Let's make this a non-allocating `Iterator` to keep the crate alloc-free. The caller can always `collect` if they want.

Reply:
Ok, I’ll look into that.

Reply:
Done. I also implemented a similar iterator for the SampleInformation blocks so that I could get rid of Vec entirely. However, it was a little more awkward because while `data_format_descriptors()` is on the top-level reader, it felt like the `sample_informations()` should be on the individual `BasicDataFormatDescriptor` structs. But those don't have the underlying file data. So, I opted to pass it into the `sample_informations(data: &[u8])`. I did try a different way first which had the BasicDataFormatDescriptor hold a borrow on the underlying data for its entire block, but that meant adding an explicit lifetime parameter to the struct which felt ugly to me.